websites/D/Desu-Online/presence.ts

```diff
@@ -0,0 +1,164 @@
+			presenceData.state = document.querySelectorAll("h1")[0].textContent;
+			presenceData.largeImageKey = document
+				.querySelector("img.ts-post-image")
+				.getAttribute("src");
+			presenceData.buttons = [
+				{
+					label: "Odwiedź",
+					url: document.URL,
+				},
+			];
+		}
+	} else if (pathname === "/harmonogram/")
+		presenceData.details = "Przegląda harmonogram...";
+	else if (pathname === "/zakladki/")
+		presenceData.details = "Przegląda swoje zakładki...";
+	else if (pathname.startsWith("/blog/")) {
+		if (pathname === "/blog/") {
+			presenceData.details = "Przegląda bloga:";
+			presenceData.state = "Strona 1";
+		} else if (pathname.startsWith("/blog/page/")) {
+			presenceData.details = "Przegląda bloga:";
+			presenceData.state = `Strona ${pathname.split("/")[3]}`;
+		} else {
+			presenceData.details = "Czyta bloga:";
+			presenceData.state = document
+				.querySelectorAll("h1")[0]
+				.textContent.replace("\n", "");
+			presenceData.buttons = [{ label: "Czytaj", url: document.URL }];
+		}
+	} else if (pathname === "/gatunki/")
+		presenceData.details = "Przegląda gatunki...";
+	else if (pathname.startsWith("/season/")) {
+		presenceData.details = "Przegląda sezon:";
+		presenceData.state = document.querySelectorAll("h1")[0].textContent;
+	} else if (pathname === "/sezony/")
+		presenceData.details = "Przegląda sezony...";
+	else if (pathname.startsWith("/label/")) {
+		if (pathname === "/label/news/")
+			presenceData.details = "Przegląda newsy...";
+		else if (pathname === "/label/recenzja/")
+			presenceData.details = "Przegląda recenzje...";
+	} else if (pathname === "/ekipa-portalu/")
+		presenceData.details = "Przegląda ekipę portalu...";
+	else if (pathname === "/polityka-prywatnosci/")
+		presenceData.details = "Przegląda politykę prywatności...";
+	else if (pathname.startsWith("/genres/")) {
+		presenceData.details = "Przegląda gatunek:";
+		presenceData.state = `${document.querySelectorAll("h1")[0].textContent}`;
+		if (pathname.includes("page")) {
+			presenceData.state = `${
+				document.querySelectorAll("h1")[0].textContent
+			} (Strona ${pathname.split("/")[4]})`;
+		}
+	} else if (pathname.startsWith("/studio/")) {
+		presenceData.details = "Przegląda studio:";
+		presenceData.state = `${document.querySelectorAll("h1")[0].textContent}`;
+		if (pathname.includes("page")) {
+			presenceData.state = `${
+				document.querySelectorAll("h1")[0].textContent
+			} (Strona ${pathname.split("/")[4]})`;
+		}
+	} else if (
+		document.querySelectorAll("span.year")[0].textContent &&
+		document
+			.querySelectorAll("span.year")[0]
+			.textContent.includes("Dodane przez")
+	) {
+		presenceData.details = `Ogląda: ${
+			document.querySelector(
+				"body > div > div > div > div > div > div > div > h3"
+			).textContent
+		}`;
+		const playinfo = document
+			.querySelector(
+				"body > div > div > div > div > div > div.episodelist > ul > li.selected > a > div.playinfo > span"
+			)
+			.textContent.split("-");
```

Comment:
More dynamic and the other can fail on some shows.
```suggestion
		const playinfo = document
			.querySelector<HTMLMetaElement>('meta[property="og:title"]')
			.content.split("-");
```

Reply:
![image](https://user-images.githubusercontent.com/42370647/194698572-38e63bcd-63da-41a4-b522-de7003b4b0c3.png)

why does it return something like this?

Reply:
if you wanted to insert this:
![image](https://user-images.githubusercontent.com/42370647/194699013-d8d923dc-943d-40f3-b66b-930f7634c16a.png)

this isnt the same as episode title:
![image](https://user-images.githubusercontent.com/42370647/194699046-5101784b-f6c0-446c-b9e7-8da981e23d52.png)
"Obietnica łez! Uprowadzony Monosuke.

Reply:
ah

Reply:
> ![image](https://user-images.githubusercontent.com/42370647/194698572-38e63bcd-63da-41a4-b522-de7003b4b0c3.png)
> 
> why does it return something like this?

the <> doesn’t work in console

Reply:
so didnt change anything here?

Reply:
> so didnt change anything here?

Recheck my previous message

Reply:
i mean, meta element doesn't include the episode title, so leave it that way?

Reply:
> i mean, meta element doesn't include the episode title, so leave it that way?

sure